extra/changelog-generator/changelog-generator

```diff
@@ -139,6 +146,73 @@ parser.add_argument(
     + "Note that for technical reasons these must come last.",
 )
 
+
+class GitQuerier:
+    def __init__(self, gitargs, base_dir):
+        self.gitargs = gitargs
+        self.base_dir = base_dir
+
+    def get_commits_for_range(self, repo, range):
+        output = subprocess.check_output(
+            ["git", "rev-list", "--reverse", range] + self.gitargs,
+            cwd=os.path.join(self.base_dir, repo),
+        )
+        return output.decode().split()
+
+    def get_raw_commit_message(self, repo, sha):
+        output = subprocess.check_output(
+            ["git", "cat-file", "-p", sha] + self.gitargs,
+            cwd=os.path.join(self.base_dir, repo),
+        )
+        # Return only the part after headers (double newline).
+        return output.decode().split("\n\n", 1)[1]
+
+    def show_commit_without_diff(self, repo, sha):
+        output = subprocess.check_output(
+            ["git", "show", "-s", sha] + self.gitargs,
+            cwd=os.path.join(self.base_dir, repo),
+        )
+        return output.decode()
+
+
+class GitHubQuerier:
+    def __init__(self):
+        # Avoid importing requests unless we have to, which is only when using
+        # this class.
+        import requests
+
+        self.requests = requests
+        self.headers = {
+            "Accept": "application/vnd.github+json",
+            "X-GitHub-Api-Version": "2022-11-28",
+        }
+        if os.getenv("GITHUB_TOKEN"):
+            self.headers["Authorization"] = "Bearer %s" % os.getenv("GITHUB_TOKEN")
+
+    def get_commits_for_range(self, repo, range):
+        range = range.replace("..", "...")
+        response = self.requests.get(
+            f"https://api.github.com/repos/mendersoftware/{repo}/compare/{range}",
+            headers=self.headers,
+        )
+        assert response.status_code == 200, f"Status code: {response.status_code}"
```

Comment:
since we're installing & importing a module already, would it not be better to go for [PyGithub](https://github.com/PyGithub/PyGithub) and then use the [compare method](https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html#github.Repository.Repository.compare) here and have access to the other utils? that would also help with keeping this up with Github API changes (which looking at their versioning scheme, _might_ be a bit more frequent...)

Reply:
Good idea! It's a very clean API, so the code is almost the same.